Objects/dictobject.c

```diff
@@ -1745,16 +1750,11 @@ insertdict(PyInterpreterState *interp, PyDictObject *mp,
     MAINTAIN_TRACKING(mp, key, value);
 
     if (ix == DKIX_EMPTY) {
-        assert(!_PyDict_HasSplitTable(mp));
```

Comment:
Was it intentional to remove this assertion? Doesn't seem related to the dict watcher stuff.

Reply:
No, thanks for catching that, I'll remove it.

Luckily there was no debugging involved! Just trying to re-factor some existing code to use dict watchers and I was worried about what would happen when the updates would fail (as the existing code handled it). So this will actually make that somewhat cleaner!